File: Base64/CPP/Base64.cpp
#include "Base64.hpp"
#include "base64/base64.h"

#include "../../Blob/CPP/Blob.hpp"
//#include "../../ExternalString/CPP/ExternalString.hpp"
//#include "../../InternalString/CPP/InternalString.hpp"

#include <cstring>

namespace NumbatLogic
{
	gsBlob* Base64::Decode(const char* sxBase64)
	{
		int nSize;
		unsigned char* pData = unbase64(sxBase64, strlen(sxBase64), &nSize);

		gsBlob* pBlob = new gsBlob();
		pBlob->PackData(pData, nSize);
		pBlob->SetOffset(0);

		return pBlob;
	}
}

File: Source/Base64/Base64.cs
using System;

namespace NumbatLogic
{
	public class Base64
	{
		public static gsBlob Decode(string sxBase64)
		{
			if (sxBase64 == null)
				return null;

			try
			{
				byte[] data = Convert.FromBase64String(sxBase64);
				gsBlob blob = new gsBlob();
				blob.Pack(data, data.Length);
				blob.SetOffset(0);
				return blob;
			}
			catch
			{
				return null;
			}
		}
	}
}
